File: download_data.py
"""Downloads csv data from the web to a local filepath as a csv.
Usage: download_data.py --url=<url> --path=<path> 
 
Options:
<url>               URL from where to download the data (must be in standard csv format)
<path>          Path (including filename) of where to locally write the file
"""

import os
import pandas as pd
from docopt import docopt

opt = docopt(__doc__)

def main(url, path):
    data = pd.read_csv(url, header=None)
    try:
        data.to_csv(path, index=False)
    except:
        os.makedirs(os.path.dirname(path))
        data.to_csv(path, index=False)



if __name__ == "__main__":
    main(opt["--url"], opt["--path"])

File: download_data.R
"Downloads csv data from the web to a local filepath as a csv.
Usage: download_data.R --url=<url> --path=<path> 
 
Options:
<url>               URL from where to download the data (must be in standard csv format)
<path>          Path (including filename) of where to locally write the file
"-> doc

library(docopt)
library(readr)

opt <- docopt(doc)

main <- function(url, path) {
  data = read_csv(url)
  write_csv(data, path)
}

main(opt$url, opt$path)
